projects/plugins/jetpack/extensions/blocks/payments-intro/edit.js

```diff
@@ -18,40 +18,42 @@ import {
 import defaultVariations from './variations';
 
 export default function JetpackPaymentsIntroEdit( { name, clientId } ) {
-	const { blockType, defaultVariation, variations, hasInnerBlocks } = useSelect( select => {
-		const { getBlockType, getBlockVariations, getDefaultBlockVariation } = select( blocksStore );
+	const { blockType, hasInnerBlocks } = useSelect( select => {
 		const { getBlocks } = select( blockEditorStore );
 
 		return {
 			blockType: getBlockType( name ),
-			defaultVariation: getDefaultBlockVariation( name, 'block' ),
-			variations: getBlockVariations( name, 'block' ),
 			hasInnerBlocks: getBlocks( clientId )?.length > 0,
 		};
 	} );
 
 	const { replaceBlock, selectBlock } = useDispatch( blockEditorStore );
 
 	const setVariation = variation => {
-		replaceBlock( clientId, createBlock( variation.name.replace( 'payments-intro', 'jetpack' ) ) );
+		replaceBlock( clientId, createBlock( variation.name ) );
 		selectBlock( clientId );
 	};
 
+	const usableVariations = unfilteredVariations => {
+		return unfilteredVariations.filter( variation => getBlockType( variation.name ) );
+	};
```

Comment:
We could make this clearer with some different wording:

```js
const registeredVariations = defaultVariations.filter( variation => getBlockType( variation.name ) );
```

(I don't think we need to have this as a function, as the parameter is always `defaultVariations` 🤔 )